src/Maestro/Maestro.Web/Api/v2020_02_20/Controllers/BuildsController.cs

```diff
@@ -363,15 +365,16 @@ private async Task SetBuildIncoherencyInfoAsync(Data.Models.Build build)
                             Commit = incoherence.Commit
                         });
                     }
+                    build = await context.Builds.FindAsync(buildId);
```

Comment:
I don't think this will retrieve the build from the db if it has been changed since retrieving it on line 347. EF remembers that it already grabed it. You should do `Context.Entry<Data.Models.Build>(build).Reload()` here.

Reply:
Just some rant.. Perhaps we can just try to persist a list of `BuildIncoherence` instead of the build and it's dependents.

Reply:
Feel free to open an issue to store these differently. I'll do the reload.